subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/artifact/DefaultVisitedArtifactResults.java

```diff
@@ -24,10 +24,13 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Function;
 
 import static org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact.ResolvedArtifactSet.EMPTY;
 
 public class DefaultVisitedArtifactResults implements VisitedArtifactsResults {
+    private static final Function<ResolvedArtifactSet, ResolvedArtifactSet> DO_NOTHING = resolvedArtifactSet -> resolvedArtifactSet;
+    private static final Function<ResolvedArtifactSet, ResolvedArtifactSet> ALLOW_UNAVAILABLE = resolvedArtifactSet -> resolvedArtifactSet instanceof UnavailableResolvedArtifactSet ? ResolvedArtifactSet.EMPTY : resolvedArtifactSet;
```

Comment:
🧐 This seems like too much for a 1-liner, debugging a failure in this would be a pain.

Reply:
I broke it up